packages/sdk-kaios/src/deviceManager.ts

```diff
@@ -1,34 +1,42 @@
-import { fsExistsSync, getRealPath, chalk, logDefault, RnvError, RnvContext } from '@rnv/core';
+import { fsExistsSync, getRealPath, chalk, logDefault, RnvError, RnvContext, logError, executeAsync, logSuccess, inquirerPrompt, getDirectories } from '@rnv/core';
+import path from 'path';
 
 const childProcess = require('child_process');
 
-export const launchKaiOSSimulator = (c: RnvContext) =>
+export const launchKaiOSSimulator = async (c: RnvContext, target: string | boolean) => {
+    logDefault(`launchKaiOSSimulator: ${target}`);
+
+    const kaiosSdkPath = getRealPath(c, c.buildConfig?.sdks?.KAIOS_SDK);
+
+    if (!kaiosSdkPath) {
+        return Promise.reject(`c.buildConfig.sdks.KAIOS_SDK undefined`);
+    }
+
+    if(target === true){
+        let availableSimulatorVersions = getDirectories(kaiosSdkPath).filter(directory=> directory.toLowerCase().indexOf("kaios") !== -1);
+
+        const { selectedSimulator } = await inquirerPrompt({
+            name: 'selectedSimulator',
+            type: 'list',
+            message: 'What simulator would you like to launch?',
+            choices: availableSimulatorVersions,
+        });
+        target = selectedSimulator
+    }
+
+    const simulatorPath = path.join(kaiosSdkPath, `${target}/kaiosrt/kaiosrt`);
+
+    if (simulatorPath && !fsExistsSync(simulatorPath)) {
+        return Promise.reject(`Can't find simulator at path: ${simulatorPath}`);
+    }
+    
     new Promise<void>((resolve, reject) => {
-        logDefault('launchKaiOSSimulator');
-
-        if (!c.buildConfig?.sdks?.KAIOS_SDK) {
-            reject(
-                `KAIOS_SDK is not configured in your ${
-                    c.paths.workspace.config
-                } file. Make sure you add location to your Kaiosrt App path similar to: ${chalk().white.bold(
-                    '"KAIOS_SDK": "/Applications/Kaiosrt.app"'
-                )}`
-            );
-            return;
-        }
-
-        const ePath = getRealPath(c, c.buildConfig?.sdks?.KAIOS_SDK);
-
-        if (ePath && !fsExistsSync(ePath)) {
-            reject(`Can't find emulator at path: ${ePath}`);
-            return;
-        }
-
-        childProcess.exec(`open ${ePath}`, (err: RnvError) => {
+        childProcess.exec(`(cd ${kaiosSdkPath}/${target}/kaiosrt && ${simulatorPath} )`, (err: RnvError) => {
```

Comment:
@Marius456 not relevant directly to your change but if you come across child_process.exec it's good practice to see if that can be converted to use `executeAsync`

Reply:
@Marius456 
```
childProcess.exec(`(cd ${kaiosSdkPath}/${target}/kaiosrt && ${simulatorPath} )`, (err: RnvError) => {
```

is equivalent to:

```
childProcess.exec(`(cd '/FULL/PATH_TO_SIM' && '/FULL/PATH_TO_SIM' )`, (err: RnvError) => {
```

is that intentional ?

Reply:
yes, because kaios simulator only opens in its directory. If you call it from outside directory, simulator opens up but it just show black screen.